graphistry/PlotterBase.py

```diff
@@ -102,17 +102,20 @@ class PlotterBase(Plottable):
     _defaultEdgeSourceId = SRC
     _defaultEdgeDestinationId = DST
     
+    # Thread-safe global caches with lock protection
+    _cache_lock = threading.RLock()
     _pd_hash_to_arrow : WeakValueDictionary = WeakValueDictionary()
     _cudf_hash_to_arrow : WeakValueDictionary = WeakValueDictionary()
     _umap_param_to_g : WeakValueDictionary = WeakValueDictionary()
     _feat_param_to_g : WeakValueDictionary = WeakValueDictionary()
```

Comment:
i don't think a client is thread safe to beginwith, so maybe the starting question is what thread safety we are trying to achieve

ex: "multithreading support is by client-per-thread"
- you can have diff clients access across diff client contexts
- a client and sub-plottables can be safely access in a thread / event loop
- access to the client / sub-plottables by other threads / event loops require manual coordination at method boundaries, e.g., locks
- you can multiple clients on different threads without requiring coordination

given such a model, i'm not sure we need this change